rewrite-java/src/main/java/org/openrewrite/java/marker/JavaVersion.java

```diff
@@ -43,4 +34,57 @@
     String vmVendor;
     String sourceCompatibility;
     String targetCompatibility;
+
+    public static MajorVersion getMajorVersion(String version) {
+        return MajorVersion.get(version);
+    }
+
+    public enum MajorVersion {
+        INVALID("invalid"),
+        JAVA_6("6"),
+        JAVA_7("7"),
+        JAVA_8("8"),
+        JAVA_9("9"),
+        JAVA_10("10"),
+        JAVA_11("11"),
+        JAVA_12("12"),
+        JAVA_13("13"),
+        JAVA_14("14"),
+        JAVA_15("15"),
+        JAVA_16("16"),
+        JAVA_17("17");
+
+        private static final Map<String, MajorVersion> lookup = new HashMap<>();
```

Comment:
I am not sure you even need a map here, we are talking about iterating over 11 values.

I would just iterate over those values: 

```
public static MajorVersion get(String version) {
    for (MajorVersion v : MajorVersion.values()) {
        if (v.value.equals(version)) {
            return v;
        }
    }
    return INVALID;
}
```

Reply:
If you wanted to, you could order the enums from most popular to least, but I just don't feel this micro-optimization buys you much.

Reply:
Fair point, thanks! I added a response to Jon's question about the enum, and will add that based on his response.

Reply:
Add the changes

Reply:
Update removed the enum per Jon's comment